crates/anvil/src/eth/api.rs

```diff
@@ -1825,6 +1829,64 @@ impl EthApi {
         Ok(())
     }
 
+    /// Deals ERC20 tokens to a address
+    ///
+    /// Handler for RPC call: `anvil_dealERC20`
+    pub async fn anvil_deal_erc20(
+        &self,
+        address: Address,
+        token_address: Address,
+        balance: U256,
+    ) -> Result<()> {
+        node_info!("anvil_dealERC20");
+
+        sol! {
+            #[sol(rpc)]
+            contract IERC20 {
+                function balanceOf(address target) external view returns (uint256);
+            }
+        }
+
+        let calldata = IERC20::balanceOfCall { target: address }.abi_encode();
+        let tx = TransactionRequest::default().with_input(calldata.clone());
+        let access_list_result = self.create_access_list(WithOtherFields::new(tx), None).await?;
+        let access_list = access_list_result.access_list;
+        for item in access_list.0 {
+            if item.address != token_address {
+                continue;
+            };
+            for slot in item.storage_keys.iter() {
+                let tx = alloy_rpc_types::transaction::TransactionRequest::default()
+                    .with_input(calldata.clone());
+
+                let account_override = alloy_rpc_types::state::AccountOverride::default()
+                    .with_state_diff(std::iter::once((*slot, B256::from(balance.to_be_bytes()))));
+
+                let mut state_override = alloy_primitives::map::AddressHashMap::default();
+                state_override.insert(token_address, account_override);
```

Comment:
there is now:

https://github.com/alloy-rs/alloy/blob/dc3feb7e3507b6fc2ab352670fcd4d58f770a7d2/crates/rpc-types-eth/src/state.rs#L12-L12

that we can use to make this a bit nicer